kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -355,7 +356,7 @@ def foundry_prove(
             setup_methods[contract.name] = f'{contract.name}.{method.name}'
 
     def run_cfg_group(tests: list[str]) -> dict[str, bool]:
-        ag_proofs: dict[str, AGProof] = {}
+        ag_proofs: dict[str, AGProof | AGBMCProof] = {}
         for test in tests:
             if AGProof.proof_exists(test, ag_proofs_dir) and not reinit:
                 ag_proof = AGProof.read_proof(test, ag_proofs_dir)
```

Comment:
what if the BMC proof already exists on disk? We should probably be cehcking "does the file exist on disk?", and then reading in the json, and inspecting the `type` field to determine if we should read a `AGProof` or `AGBMCProof`.